mathics/eval/hyperbolic.py

```diff
@@ -3,8 +3,13 @@
 """
 from mathics.core.convert.sympy import from_sympy
 
+from sympy import Symbol
 
 def eval_ComplexExpand(expr, vars):
     sympy_expr = expr.to_sympy()
-    # Turn Re[x] -> x and remove Im[x] for all variables X that are not in vars.
+    sympy_vars = {v.to_sympy() for v in vars.elements}
+    # All vars are assumed to be real
+    replaces = [(fs, Symbol(fs.name, real=True)) for fs in sympy_expr.free_symbols 
+                if fs in sympy_vars]
```

Comment:
in WMA, the vars in `vars` are assumed to be complex, so the condition would be

`if fs not in sympy_vars`